@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -250,28 +250,39 @@ export class PickHandler {
         intersect: THREE.Intersection,
         pickResult: PickResult
     ) {
+        if (featureData.objInfos === undefined) {
+            return;
+        }
+
         if (pickResult.intersection!.object instanceof MapViewPoints) {
-            pickResult.userData = featureData.objInfos![intersect.index!];
+            pickResult.userData = featureData.objInfos[intersect.index!];
             return;
-        } else if (
-            featureData.objInfos === undefined ||
+        }
+
+        if (
             featureData.starts === undefined ||
-            intersect.faceIndex === undefined
+            featureData.starts.length === 0 ||
+            (intersect.faceIndex === undefined && intersect.index === undefined)
         ) {
             return;
         }
 
-        if (featureData.starts.length > 1) {
-            let objInfosIndex = 0;
-            for (const polygonStartFace of featureData.starts) {
-                if (polygonStartFace > intersect.faceIndex * 3) {
-                    break;
-                }
-                objInfosIndex++;
-            }
-            pickResult.userData = featureData.objInfos[objInfosIndex - 1];
-        } else {
+        if (featureData.starts.length === 1) {
             pickResult.userData = featureData.objInfos[0];
+            return;
+        }
+
+        const intersectIndex =
+            intersect.faceIndex !== undefined ? intersect.faceIndex * 3 : intersect.index!;
```

Comment:
We use a face index if given, otherwise we expect a vertex index. It can only get simpler if we always use the vertex index, but can't be sure it'll be provided since it's an optional member of  THREE.Intersection, so we need to check also faceIndex to be sure.

I don't see how checking the pick type would improve this. The PickObjectType doesn't necessarily tell you anything about the picked object. If the type is Line, the object can be represented as a Line or as a Mesh.